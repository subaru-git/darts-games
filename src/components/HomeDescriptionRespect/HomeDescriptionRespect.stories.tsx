/* eslint-disable import/named */
import React from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import HomeDescriptionRespect from './HomeDescriptionRespect'

export default {
  title: 'Components/HomeDescriptionRespect',
  component: HomeDescriptionRespect,
} as ComponentMeta<typeof HomeDescriptionRespect>

const Template: ComponentStory<typeof HomeDescriptionRespect> = (args) => (
  <HomeDescriptionRespect {...args} />
)

export const Default = Template.bind({})
Default.args = {}
