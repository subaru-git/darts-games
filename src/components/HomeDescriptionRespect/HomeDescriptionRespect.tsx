import React, { FC } from 'react'
import { Center, Grid, GridItem, Text } from '@chakra-ui/react'
import YouTube from 'react-youtube'
import useLocale from '@/hooks/locale'

const HomeDescriptionRespect: FC = () => {
  const { t } = useLocale()
  return (
    <div>
      <Grid templateColumns='repeat(2, auto)' justifyContent='center' gap={4}>
        <GridItem>
          <Center h='100%'>
            <Text fontSize='xl'>{t.HOME_RESPECT}</Text>
          </Center>
        </GridItem>
        <GridItem>
          <YouTube videoId='HiNOjEYbUXc' />
        </GridItem>
      </Grid>
    </div>
  )
}

export default HomeDescriptionRespect
