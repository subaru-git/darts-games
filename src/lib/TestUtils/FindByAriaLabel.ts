export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
