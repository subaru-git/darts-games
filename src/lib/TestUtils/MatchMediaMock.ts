import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
