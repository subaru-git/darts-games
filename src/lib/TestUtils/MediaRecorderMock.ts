export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
