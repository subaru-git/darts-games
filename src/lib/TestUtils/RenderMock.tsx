import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
