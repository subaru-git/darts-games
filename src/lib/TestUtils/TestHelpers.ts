import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}