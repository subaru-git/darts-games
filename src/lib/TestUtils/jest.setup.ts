import mocksdk from 'firebase-mock';
import failOnConsole from 'jest-fail-on-console';
import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');
jest.mock('@/firebase/client', () => ({
  analytics: mocksdk.analytics,
  db: mocksdk.firestore,
  storage: mocksdk.storage,
  auth: mocksdk.auth,
  functions: mocksdk.functions,
}));
jest.mock('@firebase/auth');
beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});

failOnConsole();
