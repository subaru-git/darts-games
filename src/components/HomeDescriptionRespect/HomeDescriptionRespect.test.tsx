import { ChakraProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import { createMatchMedia } from '../../lib/TestUtils/MatchMediaMock';
import HomeDescriptionRespect from './HomeDescriptionRespect';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
test('should rendering', () => {
  window.matchMedia = createMatchMedia(1100);
  const { container } = render(
    <ChakraProvider>
      <HomeDescriptionRespect />
    </ChakraProvider>,
  );
  expect(container.getElementsByClassName('YouTube')).not.toBeNull();
  expect(container).toMatchSnapshot();
});
