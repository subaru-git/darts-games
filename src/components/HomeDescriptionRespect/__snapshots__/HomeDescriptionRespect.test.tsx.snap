// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div>
    <div
      class="css-vyd9jy"
    >
      <div
        class="css-0"
      >
        <div
          class="css-tkevr6"
        >
          <p
            class="chakra-text css-9833tc"
          >
            This site appreciates and respects all darts content, including YouTube videos.
          </p>
        </div>
      </div>
      <div
        class="css-0"
      >
        <div
          class=""
        >
          <div
            class=""
            id=""
          />
        </div>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
